docs: Add guide for local API setup, update api-usage slightly (#235)
codeRIT/hackathon-manager: docs/api-usage.md

|-------------------|--------------------------------------------------------------------------------------------------|
| List all tags     | `GET https://your-hackathon.io/manage/trackable_tags.json`                                      |
| View specific tag | `GET https://your-hackathon.io/manage/trackable_tags/1.json`                                    |
| Create tag        | `POST https://your-hackathon.io/manage/trackable_tags.json` (with body parameters)              |
| Update tag        | `PATCH https://your-hackathon.io/manage/trackable_tags/1.json` (with body parameters) |
| Delete tag        | `DELETE https://your-hackathon.io/manage/trackable_tags/1.json`                                 |

For a full list of endpoints, run `bin/rails routes` locally. This utility, provided by Ruby on Rails, lists all possible paths you can route too (along with their respective HTTP method).

> Note: Datatable endpoints are highly coupled to [Datatables](https://datatables.net) functionality and are not easy to use.

> Note: The following endpoints are currently being rewritten to allow for less restrictions, i.e., bypassing the `/manage/`-level permission.

> Example for questionnaire management endpoints:

Reviewer: I believe there should be a space between the note line and next paragraph
```suggestion

Example for questionnaire management endpoints:
```

- This "show" endpoint is the natural mapping for something like `GET /manage/questionnaires/1.json`
- Because `respond_with` is used, it will return JSON when `.json` is used in the URL

## Authentication

> Authentication is implemented with OAauth 2, provided by the [Doorkeeper](https://github.com/doorkeeper-gem/doorkeeper) gem. For full Doorkeeper endpoints + docs, see the [Doorkeeper wiki](https://github.com/doorkeeper-gem/doorkeeper/wiki/API-endpoint-descriptions-and-examples) for more info.

Reviewer: ```suggestion
Authentication is implemented with OAauth 2, provided by the [Doorkeeper](https://github.com/doorkeeper-gem/doorkeeper) gem. For full Doorkeeper endpoints + docs, see the [Doorkeeper wiki](https://github.com/doorkeeper-gem/doorkeeper/wiki/API-endpoint-descriptions-and-examples).
```

> For a guide to a local Postman/Paw setup, see the page on [API Setup](api-setup.md).

Reviewer: ```suggestion
For a guide to set up & test the API using Postman or Paw, see the [API Setup](api-setup.md) page.
```

- This "show" endpoint is the natural mapping for something like `GET /manage/questionnaires/1.json`
- Because `respond_with` is used, it will return JSON when `.json` is used in the URL

## Authentication

Authentication is implemented with OAauth 2, provided by the [Doorkeeper](https://github.com/doorkeeper-gem/doorkeeper) gem. For full Doorkeeper endpoints + docs, see the [Doorkeeper wiki](https://github.com/doorkeeper-gem/doorkeeper/wiki/API-endpoint-descriptions-and-examples).

> For a guide to set up & test the API using Postman or Paw, see the [API Setup](api-setup.md) page.

Reviewer 2: "API Setup" should probably be "Setup Postman/Paw for API Testing" to match page name
